metadata_backend/api/auth.py

```diff
@@ -146,11 +147,24 @@ async def callback(self, req: Request) -> Response:
 
         session_id = cookie["id"]
 
-        req.app["Session"][session_id] = {"oidc_state": params["state"], "access_token": access_token}
+        req.app["Session"][session_id] = {"oidc_state": params["state"], "access_token": session["token"]}
         req.app["Cookies"].add(session_id)
-        req.app["OIDC_State"].remove(params["state"])
 
-        await self._set_user(req, session_id, access_token)
+        user_data: Tuple[str, str]
+        if "eppn" in session["userinfo"]:
```

Comment:
for SDS-AAI this changed a tad see: https://github.com/CSCfi/metadata-submitter/commit/47ba615bec6a078ed18053635b0a451afa7e8643#diff-ebe74c356287148a91f17f859c1f7f9cac90af19df48b372bc1a83d30af76c06R201-R204 `CSCUserName` and `remoteUserIdentifier` might be needed now